itkwidgets/standalone_server.py

```diff
@@ -0,0 +1,387 @@
+
+    input_obj = input_dict()
+    server.register_service(
+        {
+            "name": "itkwidgets_input_obj",
+            "id": "itkwidgets-input-obj",
+            "description": "Provide the data and config object required to create a viewer.",
+            "config": {
+                "visibility": "protected",
+                "require_context": False,
+                "run_in_executor": True,
+            },
+            "inputObject": lambda: input_obj,
+        }
+    )
+
+    server.register_service(
+        {
+            "name": "itkwidgets_viewer_ready",
+            "id": "itkwidgets-viewer-ready",
+            "description": "",
+            "config": {
+                "visibility": "protected",
+                "require_context": False,
+                "run_in_executor": True,
+            },
+            "viewerReady": viewer_ready,
+        }
+    )
+
+    workspace = server.config.workspace
+    token = server.generate_token()
+    params = urlencode({"workspace": workspace, "token": token})
+    webbrowser.open_new_tab(f"{server_url}/itkwidgets/index.html?{params}")
+    return server
+
+
+def main():
+    JWT_SECRET = str(uuid.uuid4())
+    os.environ["JWT_SECRET"] = JWT_SECRET
+    hypha_server_env = os.environ.copy()
+
+    server_url = f"http://{SERVER_HOST}:{PORT}"
+    viewer_mount_dir = str(Path(VIEWER_HTML).parent)
+
+    out = None if OPTS.verbose else subprocess.DEVNULL
+    err = None if OPTS.verbose else subprocess.STDOUT
+    with subprocess.Popen(
+        [
+            sys.executable,
+            "-m",
+            "hypha.server",
+            f"--host={SERVER_HOST}",
+            f"--port={PORT}",
+            "--static-mounts",
+            f"/itkwidgets:{viewer_mount_dir}",
+        ],
+        env=hypha_server_env,
+        stdout=out,
+        stderr=err,
+    ):
+        timeout = 10
+        while timeout > 0:
+            try:
+                http = PoolManager()
+                response = http.request("GET", f"{server_url}/health/liveness")
+                if response.status == 200:
+                    break
+            except exceptions.MaxRetryError:
+                pass
+            timeout -= 0.1
+            time.sleep(0.1)
+
+        server = start_viewer(server_url)
+        if OPTS.repl:
+            EVENT.wait()  # Wait until viewer is created before launching REPL
+            workspace = server.config.workspace
+            svc = server.get_service(f"{workspace}/itkwidgets-client:itk-vtk-viewer")
+            viewer = view(itk_viewer=svc.viewer())
+            banner = f"""
+                Welcome to the itkwidgets command line tool! Press CTRL+D or
+                run `exit()` to terminate the REPL session. Use the `viewer`
+                object to manipulate the viewer.
+            """
+            exitmsg = "Exiting REPL. Press CTRL+C to teminate CLI tool."
+            code.interact(banner=banner, local={"viewer": viewer}, exitmsg=exitmsg)
+
+
```

Comment:
Can we also add:

```
if __name__ == "__main__":
    cli_entrypoint()
```

?

Reply:
Sure! But should this be
```
if __name__ == "__main__":
  cli_entrypoint()
```
instead so that we still parse any flags?

Reply:
yes :-)